app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchViewHolder.kt

```diff
@@ -20,41 +23,53 @@ class SearchViewHolder(
-    private val bottomTextView = binding.bottomTextView
-
     fun onBind(item: SearchResult) {
         when (item) {
-            is SearchResult.Full -> {
-                if (item.username.endsWith(usernameDomainFeatureToggle.value)) {
-                    binding.walletImageView.background = null
-                    binding.walletImageView.setImageResource(R.drawable.ic_key_app_circle)
-                } else {
-                    binding.walletImageView.background = binding.getDrawable(R.drawable.bg_app_rounded)
-                    binding.walletImageView.setImageResource(R.drawable.ic_wallet_gray)
-                }
-
-                topTextView.text = item.username
-                bottomTextView.withTextOrGone(item.addressState.address.cutEnd())
-                bottomTextView.setTextColor(bottomTextView.context.getColor(R.color.backgroundDisabled))
-            }
-            is SearchResult.AddressOnly -> {
-                topTextView.text = item.addressState.address.cutEnd()
-                bottomTextView.isVisible = false
-            }
-            is SearchResult.EmptyBalance -> {
-                topTextView.text = item.addressState.address.cutEnd()
-                val caution = bottomTextView.context.getString(R.string.send_caution_empty_balance)
-                bottomTextView.withTextOrGone(caution)
-                val warningColor = bottomTextView.context.getColor(R.color.systemWarningMain)
-                bottomTextView.setTextColor(warningColor)
-            }
-            is SearchResult.Wrong -> {
-                Timber.w("Received SearchResult.Wrong in unexpected place")
-                // do nothing, no wrong type should be in search view
-            }
+            is SearchResult.UsernameFound -> renderFull(item)
+            is SearchResult.AddressOnly -> renderAddressOnly(item)
+            is SearchResult.EmptyBalance -> renderEmptyBalance(item)
+            // do nothing, no wrong type should be in search view
+            is SearchResult.InvalidAddress -> Timber.w("Received SearchResult.Wrong in unexpected place")
         }
 
         itemView.setOnClickListener { onItemClicked(item) }
     }
+
+    private fun renderFull(item: SearchResult.UsernameFound) {
+        @DrawableRes
+        val imageResource: Int
+        with(binding) {
+            if (item.username.endsWith(usernameDomainFeatureToggle.value)) {
+                walletImageView.background = null
+                imageResource = R.drawable.ic_key_app_circle
+            } else {
+                walletImageView.background = getDrawable(R.drawable.bg_app_rounded)
+                imageResource = R.drawable.ic_wallet_gray
+            }
+
+            Glide.with(root)
+                .load(imageResource)
+                .circleCrop()
+                .into(walletImageView)
+
+            topTextView.text = item.username
+            bottomTextView.withTextOrGone(item.addressState.address.cutEnd())
```

Comment:
I think it's simpler to remove the brackets for better readability. 
the function is especially created as `infix` to use it without brackets

what do you think?
`bottomTextView withTextOrGone item.addressState.address.cutEnd()`

Reply:
i even don't like this method, it created a bug for me
but i'll remove brackets only for you ;)